java/statestore/src/test/java/sleeper/statestore/transactionlog/TransactionLogStateStoreDynamoDBSpecificIT.java

```diff
@@ -216,43 +224,59 @@ void shouldLoadLatestFilesSnapshotIfNoPartitionsSnapshotIsPresent() throws Excep
+            addFilesForSnapshot(files, 1);
+            createSnapshot();
 
             // When
-            StateStore stateStore = createStateStore();
+            StateStore stateStore = stateStore();
 
             // Then
             assertThat(stateStore.getAllPartitions()).isEmpty();
             assertThat(stateStore.getFileReferences())
-                    .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
                     .containsExactlyElementsOf(files);
         }
-    }
 
-    private void saveFilesSnapshot(List<FileReference> files, long transactionNumber) throws Exception {
-        StateStoreFiles state = new StateStoreFiles();
-        files.forEach(file -> state.add(fileWithOneReference(file, Instant.now())));
-        TransactionLogSnapshot snapshot = TransactionLogSnapshot.forFiles(tempDir.toString(), transactionNumber);
-        snapshotSerDe.saveFiles(snapshot, state);
-        snapshotStore.saveSnapshot(snapshot);
-    }
+        private void initialisePartitionsForSnapshot(PartitionTree tree, long transactionNumber) throws DuplicateTransactionNumberException {
+            partitionsStore.addTransaction(new TransactionLogEntry(transactionNumber, DEFAULT_UPDATE_TIME,
+                    new InitialisePartitionsTransaction(tree.getAllPartitions())));
+        }
+
+        private void addFilesForSnapshot(List<FileReference> files, long transactionNumber) throws DuplicateTransactionNumberException {
+            filesStore.addTransaction(new TransactionLogEntry(transactionNumber, DEFAULT_UPDATE_TIME,
+                    new AddFilesTransaction(files.stream()
+                            .map(file -> AllReferencesToAFile.fileWithOneReference(file, DEFAULT_UPDATE_TIME))
+                            .collect(Collectors.toList()))));
+        }
+
+        private void createSnapshot() {
+            new TransactionLogSnapshotCreator(
+                    instanceProperties, tableProperties, filesStore, partitionsStore, dynamoDBClient, configuration)
+                    .createSnapshot();
+        }
 
-    private void savePartitionsSnapshot(PartitionTree partitionTree, long transactionNumber) throws Exception {
-        StateStorePartitions state = new StateStorePartitions();
-        partitionTree.getAllPartitions().forEach(state::put);
-        TransactionLogSnapshot snapshot = TransactionLogSnapshot.forPartitions(tempDir.toString(), transactionNumber);
-        snapshotSerDe.savePartitions(snapshot, state);
-        snapshotStore.saveSnapshot(snapshot);
+        private FileReferenceFactory factory(PartitionTree tree) {
+            return FileReferenceFactory.fromUpdatedAt(tree, DEFAULT_UPDATE_TIME);
+        }
+
+        private StateStore stateStore() {
+            StateStore stateStore = createStateStore();
+            stateStore.fixFileUpdateTime(DEFAULT_UPDATE_TIME);
+            stateStore.fixPartitionUpdateTime(DEFAULT_UPDATE_TIME);
+            return stateStore;
```

Comment:
It looks like this method isn't needed, since it has the same code as createStateStore.